### PR TITLE
thanos-querier: proxies initial commit

### PR DESCRIPTION
**Pull request by s-urbaniak**
This adds kube-rbac-proxy/oauth proxy to thanos, and configures an external route. This builds on top of https://github.com/openshift/cluster-monitoring-operator/pull/499.

/cc @brancz @LiliC @paulfantom

### CODE REVIEW

**Comment by s-urbaniak** [comment]
setting as WIP as I'd like to get rid of the open http port first.

File: jsonnet/prometheus-operator.jsonnet [hunk 1]
```diff
@@ -1,5 +1,11 @@
 {
-  prometheusOperator+:: {
+  clusterPrometheusOperator+:: $.prometheusOperator + {
+    '0alertmanagerCustomResourceDefinition':: {},
```
Comment (brancz) [inline jsonnet/prometheus-operator.jsonnet]: this is probably worth a comment

File: pkg/operator/operator.go [hunk 2]
```diff
@@ -291,22 +292,24 @@ func (o *Operator) sync(key string) error {
...
-			tasks.NewTaskSpec("Updating prometheus-adapter", tasks.NewPrometheusAdapterTaks(o.namespace, o.client, factory)),
-			tasks.NewTaskSpec("Updating Telemeter client", tasks.NewTelemeterClientTask(o.client, factory, config.TelemeterClientConfig)),
-			tasks.NewTaskSpec("Updating configuration sharing", tasks.NewConfigSharingTask(o.client, factory)),
+			//tasks.NewTaskSpec("Updating Prometheus Operator", tasks.NewPrometheusOperatorTask(o.client, factory)),
+			tasks.NewTaskSpec("Updating userworkload Prometheus Operator", tasks.NewPrometheusOperatorUserworkloadTask(o.client, factory, config.UserworkloadConfig)),
+			//tasks.NewTaskSpec("Updating Cluster Monitoring Operator", tasks.NewClusterMonitoringOperatorTask(o.client, factory)),
+			//tasks.NewTaskSpec("Updating Grafana", tasks.NewGrafanaTask(o.client, factory)),
+			//tasks.NewTaskSpec("Updating Prometheus-k8s", tasks.NewPrometheusTask(o.client, factory, config)),
+			tasks.NewTaskSpec("Updating Prometheus-userworkload", tasks.NewPrometheusUserworkloadTask(o.client, factory, config.UserworkloadConfig)),
+			//tasks.NewTaskSpec("Updating Alertmanager", tasks.NewAlertmanagerTask(o.client, factory)),
```
Comment (brancz) [inline pkg/operator/operator.go]: I assume these will be uncommented again? :)

Comment (s-urbaniak) [reply pkg/operator/operator.go]: ooops, indeed :)

**Comment by s-urbaniak** [comment]
@brancz very good idea re: e2e tests! :+1: I wanted to tackle those individually but this is indeed a low hanging fruit.

**Comment by s-urbaniak** [comment]
@brancz @paulfantom ptal, looking good from your side or is there anything to add here?

**Comment by brancz** [comment]
We typically have unit tests at least testing the "factory" methods. They've demonstrated small mistakes very early in the past. I think they're worth adding but happy to have that done in a follow up.

/lgtm

**Comment by s-urbaniak** [comment]
@brancz indeed, I _knew_ i forgot something ;-) yes, i am working on a more elaborate e2e test, i'll add them in that commit, good catch :+1:

**Pull request merged**